### ISSUE TITLE
allow localhost/test-ip on AuthenticationContext 

### ISSUE THREAD

**Issue by oreporan** [AzureAD/azure-activedirectory-library-for-nodejs#163]
We have blackbox tests where we want to mock the ActiveDirectory. 
To do that we start a fake-server on localhost and intercept activeDirectory requests (`oauth2/token/`). 
But the AuthenticationContext throws an error since its not a known ActiveDirectory.

Suggestion:
add method to allow adding known IPs

```js
const context = new AuthenticationContext(activeDirectoryUrl || authorityUrl, false);
context.addKnownHost('http://localhost')
```
Or if this method is an overkill, just allow 'localhost' by default..

**Comment by ncjones** [comment]
I already implemented #218 before seeing this issue. Given the age of this I guess there's little to no hope that our changes will be merged. In the mean time my workaround it to apply the following monkey patch:

```javascript
// my-lib/adal.js:

const adalAuthority = require('adal-node/lib/authority');

class AuthorityWrapper extends adalAuthority.Authority {

    constructor(authorityUrl) {
        super(authorityUrl);
        /* pre initialize endpoint urls without forcing https */
        this._tokenEndpoint = `${authorityUrl}/oauth2/token`;
        this._deviceCodeEndpoint = `${authorityUrl}/oauth2/devicecode`;
    }

    _validateAuthorityUrl() {
        /* skip url validation */
    }
}

adalAuthority.Authority = AuthorityWrapper;

// require('adal-node') MUST come last so it is initialized with the patched authority 
module.exports = require('adal-node');
```

**Comment by jmckennon** [comment]
Thanks for sharing your workaround--we appreciate the help. Sorry for the lack of support; current authentication work from Microsoft is delivered through the `msal js` library [here](https://github.com/AzureAD/microsoft-authentication-library-for-js) 

There is plans to add a similar node.js wrapper to `msal js`, as can be seen on our roadmap [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki#Roadmap). `adal js`, and the adal-node wrapper, are still supported only for security fixes. We recommend moving to `msal js` for any advanced feature requests and bugfixes.
